presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java

```diff
@@ -103,7 +103,7 @@ public String getPartitionName()
 
     private Optional<InternalHiveSplit> createInternalHiveSplit(LocatedFileStatus status, OptionalInt bucketNumber, boolean splittable)
     {
-        splittable = splittable && isSplittable(inputFormat, fileSystem, status.getPath());
+        splittable &= isSplittable(inputFormat, fileSystem, status.getPath());
```

Comment:
This no longer short-circuits. `&&=` would... if it existed.
If your intention is to invoke `isSplittable` unconditionally, add a comment in the code.

Reply:
Good point. Reverted